run/test/specs/disappear_after_send_off_1o1.spec.ts

```diff
@@ -45,51 +48,54 @@ async function disappearAfterSendOff1o1(platform: SupportedPlatformsType) {
     device1,
     device2,
     time,
-    controlMode
+    controlMode,
+    device3
   );
-  // Send message to verify that deletion is working
-  await device1.sendMessage(testMessage);
-  // Check message appears on both device 2 and linked device 3
+
+  // ]);
+  // Turned off disappearing messages on device 1
+  await device1.clickOnElementAll({ strategy: 'accessibility id', selector: 'More options' });
+  await device1.clickOnElementAll(new DisappearingMessagesMenuOption(device1));
+  await device1.clickOnElementAll(new DisableDisappearingMessages(device1));
+  await device1.clickOnElementAll(new SetDisappearMessagesButton(device1));
+  await runOnlyOnIOS(platform, () => device1.navigateBack(platform));
+  // Check control message for turning off disappearing messages
+  // Check USER A'S CONTROL MESSAGE on device 1 and 3 (linked device)
+  const disappearingMessagesTurnedOffYou = englishStripped(
+    'disappearingMessagesTurnedOffYou'
+  ).toString();
+  // Check USER B'S CONTROL MESSAGE
+  const disappearingMessagesTurnedOff = englishStripped('disappearingMessagesTurnedOff')
+    .withArgs({ name: userA.userName })
+    .toString();
   await Promise.all([
-    device2.waitForTextElementToBePresent({
-      strategy: 'accessibility id',
-      selector: 'Message body',
-      text: testMessage,
-    }),
-    device3.waitForTextElementToBePresent({
-      strategy: 'accessibility id',
-      selector: 'Message body',
-      text: testMessage,
-    }),
+    device1.disappearingControlMessage(disappearingMessagesTurnedOffYou),
+    device2.disappearingControlMessage(disappearingMessagesTurnedOff),
+    // device3.disappearingControlMessage(disappearingMessagesTurnedOffYou),
   ]);
-  // Wait for message to disappear
-  await sleepFor(30000);
+  // Follow setting on device 2
+  await device2.clickOnElementAll(new FollowSettingsButton(device2));
+  await sleepFor(500);
+  await device2.checkModalStrings(
+    'disappearingMessagesFollowSetting',
+    'disappearingMessagesFollowSettingOff',
+    true
+  );
+  await device2.clickOnElementAll({ strategy: 'accessibility id', selector: 'Confirm' });
+  // Check conversation subtitle?
   await Promise.all([
-    device1.hasElementBeenDeleted({
-      strategy: 'accessibility id',
-      selector: 'Message body',
-      text: testMessage,
+    device1.doesElementExist({
+      ...new DisappearingMessagesSubtitle(device1).build(),
+      maxWait: 500,
     }),
-    device2.hasElementBeenDeleted({
-      strategy: 'accessibility id',
-      selector: 'Message body',
-      text: testMessage,
+    device2.doesElementExist({
+      ...new DisappearingMessagesSubtitle(device2).build(),
```

Comment:
you don't want to allow the types not built in doesElementExist?

Reply:
This was to allow a custom maxWait. The current implementation doesn't allow the new locator type with any extra variables. I tried a bunch of different ways and this was the best I could come up with